Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityInteractionMapping.cs

```diff
@@ -401,22 +401,9 @@ public Vector2 Vector2Data
                     Debug.LogError($"SetVector2Value is only valid for AxisType.DualAxis InteractionMappings\nPlease check the {inputType} mapping for the current controller");
                 }
 
-                if (invertXAxis || invertYAxis)
-                {
-                    float invertXAxisFactor = invertXAxis ? -1f : 1f;
-                    float invertYAxisFactor = invertYAxis ? -1f : 1f;
-
-                    Changed = !vector2Data.x.Equals(value.x * invertXAxisFactor) &&
-                              !vector2Data.y.Equals(value.y * invertYAxisFactor);
-
-                    vector2Data.x = value.x * invertXAxisFactor;
-                    vector2Data.y = value.y * invertYAxisFactor;
-                }
-                else
-                {
-                    Changed = vector2Data != value;
-                    vector2Data = value;
-                }
+                Vector2 newValue = value * new Vector2(invertXAxis ? -1f : 1f, invertYAxis ? -1f : 1f);
```

Comment:
```suggestion
               var newValue = value;
               
               if (invertXAxis)
               {
                   newValue.x *= -1f;
               }
               
               if (invertYAxis)
               {
                   newValue.y *= -1f;
               }
```

Reply:
What's the reasoning behid the proposed change?